New resource - `azurerm_servicebus_namespace_network_rule_set` (#6379)
terraform-providers/terraform-provider-azurerm: website/docs/r/servicebus_namespace_network_rule_set.html.markdown

  service_endpoints = ["Microsoft.ServiceBus"]
}

resource "azurerm_servicebus_namespace_network_rule_set" "example" {
  namespace_name      = azurerm_servicebus_namespace.example.name
  resource_group_name = azurerm_resource_group.example.name

  default_action = "Deny"

  network_rules {
    subnet_id                            = azurerm_subnet.example.id
    ignore_missing_vnet_service_endpoint = false
  }

  ip_rules = ["1.1.1.1"]
}
```

## Argument Reference

The following arguments are supported:

* `resource_group_name` - (Required) Specifies the name of the Resource Group where the ServiceBus Namespace Network Rule Set should exist. Changing this forces a new resource to be created.

* `namespace_name` - (Required) Specifies the ServiceBus Namespace name to which to attach the ServiceBus Namespace Network Rule Set. Changing this forces a new resource to be created.

~> **NOTE:** The ServiceBus Namespace must be `Premium` in order to attach a ServiceBus Namespace Network Rule Set.

* `default_action` - (Optional) Specifies the default action for the ServiceBus Namespace Network Rule Set. Possible values are `Allow` and `Deny`. Defaults to `Deny`.

> * `ip_rules` - (Optional) A list of IP filter masks that are added to allow access to the ServiceBus Namespace.

Reviewer: Based on the `network_acls` of `azurerm_key_vault` and the assumption CIDR Blocks work as well:
```suggestion
* `ip_rules` - (Optional) One or more IP Addresses, or CIDR Blocks which should be able to access the ServiceBus Namespace.
```

Author: Done

> * `network_rules` - (Optional) One or more `network_rules` blocks defined below.

Reviewer: Small fix
```suggestion
* `network_rules` - (Optional) One or more `network_rules` blocks as defined below.
```

Author: Done

---

A `network_rules` block supports the following:

> * `subnet_id` - (Required) The ID of the subnet you want to allow to access the corresponding ServiceBus Namespace.

Reviewer: Just a suggestion, also based on `azurerm_key_vault`
```suggestion
* `subnet_id` - (Required) The Subnet ID which should be able to access this ServiceBus Namespace.
```

Author: Done
